### ISSUE TITLE
JUserHelper::getProfile does not return a profile 

### ISSUE THREAD

**Issue by hermodo** [joomla/joomla-platform#343]
This is platform issue of : [ joomla Joomla! CMS Issue Tracker Item #25431] JUserHelper::getProfile does not return a profile (trackeritem-25431)
 At line 222 in libraries/joomla/user/helper.php

$data->id=$userId;
$results = $dispatcher->trigger('onContentPrepareData', array('com_users.profile', &$data));

**Comment by eddieajau** [comment]
Would you be able to organise someone to make a pull request for this change?

Thanks in advance.

**Comment by hermodo** [comment]
I'll try to see if some of the people that commented on this on the old issue tracker is willing.

**Comment by gnomeontherun** [comment]
I'll tackle this, just curious though what object should be returned by this method?

**Comment by gnomeontherun** [comment]
I submitted a pull request to address this

**Comment by hermodo** [comment]
The following will fix it.

```
function getProfile($userId = 0)
{
    if ($userId == 0) {
        $user   = JFactory::getUser();
        $userId = $user->id;
    }
    else {
        $user   = JFactory::getUser((int) $userId);
    }

    // Get the dispatcher and load the user's plugins.
    $dispatcher = JDispatcher::getInstance();
    JPluginHelper::importPlugin('user');

    $data = new JObject;
    // Bug fix for profile
    $data->id=$userId;

    // Trigger the data preparation event.
    //$results = $dispatcher->trigger('onPrepareUserProfileData', array($userId, &$data));
    // Bug fix for profile
    $results = $dispatcher->trigger('onContentPrepareData', array('com_users.profile', &$data));

    return $data;
}
```

Hermod

**Comment by gnomeontherun** [comment]
Thanks, I had basically the same thing, but removed the $user variable because its not needed. See the pull request for the actual code, but waiting for it to be adopted.

**Comment by gnomeontherun** [comment]
Request was pulled and is now in the Platform. This can be closed.

**Comment by chdemko** [comment]
Fixed in #558
